tests/controllers/TestObserverController.in.cpp

```diff
@@ -93,54 +139,93 @@ struct MC_CONTROL_DLLAPI TestObserverController : public MCController
+    BOOST_REQUIRE_EQUAL(firstPipeline.observers().size(), 4);
+    BOOST_REQUIRE(firstPipeline.hasObserver("Encoder"));
+    BOOST_REQUIRE(firstPipeline.hasObserver("Encoder2"));
+    BOOST_REQUIRE(firstPipeline.hasObserver("BodySensor"));
+    BOOST_REQUIRE(firstPipeline.hasObserver("KinematicInertial"));
+    const auto & observers = firstPipeline.observers();
+    // Check that order is respected
+    BOOST_REQUIRE_EQUAL(observers[0].observer().name(), "Encoder");
+    BOOST_REQUIRE_EQUAL(observers[0].update(), true);
+    // Check that multiple observers of same type can be loaded
+    BOOST_REQUIRE_EQUAL(observers[1].observer().name(), "Encoder2");
+    BOOST_REQUIRE_EQUAL(observers[1].update(), false);
+    BOOST_REQUIRE_EQUAL(observers[2].observer().name(), "BodySensor");
+    BOOST_REQUIRE_EQUAL(observers[2].update(), false);
+    BOOST_REQUIRE_EQUAL(observers[3].observer().name(), "KinematicInertial");
+    BOOST_REQUIRE_EQUAL(observers[3].update(), true);
+    BOOST_REQUIRE(firstPipeline.hasObserverType("Encoder"));
+    BOOST_REQUIRE(firstPipeline.hasObserverType("BodySensor"));
+    BOOST_REQUIRE(firstPipeline.hasObserverType("KinematicInertial"));
+
+    // The pipelines haven't yet been exectued in the reset function
+    BOOST_REQUIRE_EQUAL(firstPipeline.success(), false);
+
+    const auto & secondPipeline = observerPipeline("SecondPipeline");
+    BOOST_REQUIRE_EQUAL(secondPipeline.observers().size(), 3);
+    BOOST_REQUIRE_EQUAL(secondPipeline.success(), false);
+    BOOST_REQUIRE(secondPipeline.hasObserver("EncoderWithName"));
+    BOOST_REQUIRE(secondPipeline.hasObserver("SecondRobotEncoder"));
+    BOOST_REQUIRE(secondPipeline.hasObserver("SecondRobotBodySensor"));
+    BOOST_REQUIRE(secondPipeline.hasObserverType("Encoder"));
+    BOOST_REQUIRE(secondPipeline.hasObserverType("BodySensor"));
+    BOOST_REQUIRE_EQUAL(secondPipeline.observer("EncoderWithName").update(), false);
+    BOOST_REQUIRE_EQUAL(secondPipeline.observer("SecondRobotBodySensor").update(), true);
+    BOOST_REQUIRE_EQUAL(secondPipeline.observer("SecondRobotEncoder").update(), true);
+
+    // Check that the real robot was properly initialized
+    BOOST_REQUIRE(allclose(realRobot().posW().translation(), robot().posW().translation()));
+    BOOST_REQUIRE(allclose(realRobot().posW().rotation(), robot().posW().rotation()));
+
+    // Add anchor frame
+    datastore().make_call("KinematicAnchorFrame::" + robot().name(), [](const mc_rbdyn::Robot & robot) {
+      return sva::interpolate(robot.surfacePose("LeftFoot"), robot.surfacePose("RightFoot"), 0.5);
+    });
+  }
 
-    auto hasObserver = [this](const std::string & name) {
-      for(const auto & obs : observers_)
-      {
-        if(obs->name() == name)
-        {
-          return true;
-        }
-      }
-      return false;
-    };
-    auto hasPipelineObserver = [this](const std::string & name, bool update) {
-      for(const auto & obs : pipelineObservers_)
+  void simulateSensors()
```

Comment:
To mimic the behaviour of the framework more accurately I think this could be done in the test ticker